docs/cache-provider.md

```diff
@@ -47,3 +47,7 @@ render(
   </CacheProvider>
 )
 ```
+
+## Using CacheProvider with Server-Side-Rendering
+
+Emotion works with SSR out of the box. However, when implementing `CacheProvider`, we must also create our own server implementation using [`createEmotionServer`](https://emotion.sh/docs/create-emotion-server) and pass it the same `cache` to ensure styles are created and hydrated properly.
```

Comment:
[This README](https://github.com/emotion-js/emotion/tree/a46e2aa507d1da0e1b77cc648acc9cba2cd816e2/packages/create-emotion-server) has to be tweaked too as it mentions only one reason to use this API, but it also should mentioned what you have discovered here.